### DISCUSSION TITLE
The notetaker industry is moving really fast

### DISCUSSION THREAD

**Discussion by quesada** [zim-desktop-wiki/zim-desktop-wiki#2041] [Ideas]
I know @jaap-karssenberg is mostly pushing the product alone and that there's years of issues in the tracker. And I love zim, was a devout user for many years. 
I want to bring the attention of the community to one fact: There's a cambrian explosion in notetaking right now. Roam opened pandora's box and apps now all seem to have a similar (and very big!) feature set, including backlinks, graphs, etc.

Particularly logseq (open source) is moving really fast with features like flashcards, presentations, reading and annotating pdfs in-app, mobile app...

While I understand zim is not trying to compete and just build the best desktop app possible, these other notetakers build on electron and benefit from js libraries that do a lot. At the current pace of development, the unfathomable truth is that zim will become obsolete. I don't have much time to test apps, but for something as crucial as notetaking (an app I use all day every day) I did and had to abandon zim.

This is an uncomfortable truth, and perhaps @jaap-karssenberg is aware of the state of the market. Perhaps a good strategy would be to focus on one or two features that make zim different and go really deep on those.

One interesting way to look at the market, feature by feature is https://www.noteapps.info/

**Comment by sojusnik** [comment]
Focus on the essentials seems to be the only viable solution for Zim, because feature-wise it can't compete with Obsidian et al. I'm with @quesada on this one.

**Comment by reagle** [comment]
BTW: I continue to use Zim because it is cross-platform, can export to HTML, and, most importantly, has an amazing task-list plugin -- and an even more powerful one is in the pipeline I believe. A widely use markdown based app might tempt me, but I haven't seen anything with Zim's tasks: before and after date specifications, priorities, and tags.

**Reply by sojusnik** [reply]
I think that Zim would profit from markdown support a lot. This is one of the main reasons people turn to other note-taking apps.

**Reply by reagle** [reply]
The markdown work is underway. Some [refactoring needs to be done first](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1879#issuecomment-1019532070), however, there is even [a branch](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/26#issuecomment-514371780) with a hack for this.

**Reply by quesada** [reply]
I tried living with the branch that does markdown. Did a lot of massaging to survive the bugs. But at the end it was more trouble than it's worth.

Mind you other tools have problems with serializing notes in md. Logseq creates an abomination of nested bullet points that is not really standard markdown either.

**Reply by quesada** [reply]
I would love to stay in zim as the typing latency is superior to anything electron. But the downsides now are too big for me, YMMV

**Reply by sojusnik** [reply]
Yeah, the latency and bloat of electron apps is one of the major hindrances, so Zim could focus on speed as one of its main strengths.

**Comment by jaap-karssenberg** [comment]
To be honest, I would be really happy if there are better applications with bigger support teams behind it that take the same niche. If a good open source solution comes close in features I might even be tempted to stop development of Zim. However my main dedication is to have a stable application that is both open source and an open, local data format such that in fact you are able to take your data with you.

Yes, I agree zim should move to MarkDown as well - if only to allow migration between Zim and other apps ;)

**Reply by rzelnik** [reply]
It looks like there's no full-featured replacement for Zim at the moment. If you try to filter the alternative apps in [NoteApps](https://www.noteapps.info/), when you add "open source" as a desired feature, it will show [Joplin](https://joplinapp.org/), [Logseq](https://logseq.com) and [Standard Notes](https://standardnotes.org/).

- **Joplin** stores pages in Sqlite instead of text files and uses hash-based links between pages. It is not fully wysiwyg, it uses two panes - one for editing in source code and one for rendering.
- **Logseq** stores page data strictly in nested lists (like Workflowy or Roam Research), therefore it is nearly impossible to transfer data from Zim without damage.
- **Standard Notes** uses plain text only in a free version.

I tried several note-taking apps and Zim is a clear winner for me.

@jaap-karssenberg , I would like to ask you something personal: Do you like development of Zim? Do you still want to keep the project alive? Do you feel happy when working on Zim, or do you feel it just a burden and a duty? At a certain stage of development of any open source project there is a moment when a founder should become less a developer and more a project manager and a team leader. Would you like to evolve your role in the project in this sense?

**Reply by mikro2nd** [reply]
I've tried almost all of the competition, and I just stick with Zim. Why? It has all the features (backlinks, note-graph) that other tout as their Big Thing, and Thank All God its NOT markdown. Markdown is fine for web-documents, but has several significant deficiencies for wiki markup. I get that it's the Shiny Toy Of The Day, so people want to be seen wanting it, but I want better wikification (if that's a word.)

Then, too, I don't want my notes (and there's like 5-years worth) hidden in someone else's silo (Roam, Notion). There's a lot of very personal stuff in there; I'm happiest that it's on my own computers, under my own control. Further, as mentioned by @rzelnik above, storage is Just Text Files. It's change/future proof out of the box.

Admittedly my use-case is one with no requirement for mobile use.

To in total contradistinction to the origin posts "I don't have much time to test apps, but for something as crucial as notetaking (an app I use all day every day) I did and had to abandon zim" I'll reply: I, too, don't have much time to test apps, but for something as crucial as notetaking (an app I use all day every day) I tried almost all of the others and Zim is still better by far for my uses.

**Reply by rand00** [reply]
Saving notes as text-files is essential for me too:
* I can encrypt all my data with `encfs` and still have my cloud-synchronization tool of choice, `unison`, sync the diff of all my changes - because the encrypted files are still files.
* I can `diff` changes between zim files from the cloud and locally if I hit a merge-conflict when synchronizing. 
* I have the freedom to do anything with the data; even work on it without access to the zim UI. Also important for me: to be able to [query files on the filesystem](https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/2948), and exploit that zim maps the tree of notes to the filesystem - so I can query related parts of the tree by querying recursively within a directory.

**Comment by patvdv** [comment]
I think you are forgetting that Zim is actually free - free as in beer and free as in speech. All the other apps mentioned on that particular list just peddle the usual pay-by-month subscription bullshit. At the end all they want is to lock you in for the longest time possible. You have to compare apples with apples.

**Reply by sojusnik** [reply]
That's certainly not the case with [Obsidian](https://www.noteapps.info/apps/obsidian) or [Logseq](https://www.noteapps.info/apps/logseq), to name a few.

**Reply by patvdv** [reply]
On the Logseq page I read:

_Free. Pro plan coming soon._

Also Obsidian requires payments for some advanced features.

Paid apps aren't necessarily bad, it just depends on what you are looking for and what you are willing to pay. My point is that you shouldn't run down Zim simply because there are more _fancy_ freemium/premium apps out there. The whole notion that Zim is/should become obsolete because of that is just ridiculous.

**Reply by sojusnik** [reply]
I'm not saying that Zim is obsolete, it has its strengths and should build on them to stay relevant, but I get your point.

**Reply by mtill** [reply]
I tried several alternatives and spend quite some hours evaluating them. none was as good as zim in 2022. I use zim every day both personally and also for work.

**Reply by zogommog** [reply]
`All the other apps mentioned on that particular list just peddle the usual pay-by-month subscription bullshit.`

^ This says it for me, plus I'm sick to death of everything being based on Javascript/web based. A decision that the whole sum of computing should be over HTTP for everything is not one ultimately that benefits the users. I'll take a real application any day. It speaks worlds that the author uses their own code, and engages directly with people at a personal level. Is Zim it perfect? No, although that is subjective. I use it enough to know the little quirks and could list a few things to be tweaked from years of testing (I'm not up for it this moment), but it gets the job done. Thanks, Jaap for your generosity to the world.

If I were to make a switch, and I'm skeptical it would be worth my time, I'd only consider something that was more of an apples-to-apples comparison (not a web page based application, no strings attached). From what I can tell, the most interesting other thing out there that is comparable is CherryTree, oddly not even mentioned.

**Reply by rzelnik** [reply]
JS/web-based applications have an advantage of easy multi-platform development, which is difficult to achieve with native code. I assume this is why we still do not have mobile versions of Zim.

**Reply by mtill** [reply]
@rzelnik try Markor - it supports the ZIM wiki format on Android

**Reply by rzelnik** [reply]
@mtill I know Markor and I aready use it.

**Comment by UnisTorvalds** [comment]
I disagree.  Zim does not compete with Roam, Logseq and the like because Zim is not a notetaking app.  Although you could use it for note taking, Zim is fundamentally a wiki.  It's a different thing.
Rather than comparing with Joplin, Evernote, etc., Zim would more properly be compared with MediaWiki, DokuWiki, XWiki, Gollum, etc..
And in that regard, Zim is utterly unique, as it is the only one (afaik) optimized for a single-user offline (desktop) experience.

I use both Zim and Joplin every day and they fulfill very different use-cases for me.

**Reply by rzelnik** [reply]
@Heston-in-Mtl , what's the fundamental difference between a note-taking app and a single-user wiki in your opinion?

**Reply by quesada** [reply]
Really interesting opinions, everyone.
I think single player wiki gets really close to notetaking app nowadays. Mostly because state of the art notetaking (eg not Joplin, evernote etc) uses [[wikilinks]] for everything. To that they add backlinks, graph view etc. 
What is really amazing with zim is that it's a real desktop app, and everything is really fast. This is worth a lot to me.

The current tool I'm using (and that has a fantastic new paradigm for seeing note relations, better than the useless graph view of Roam variants) is Tangent: http://tangentnotes.com/

The sliding windows paradigm is revolutionary too to keep the train of thought. It's the one app that I can tell made me better after adopting it.

**Reply by mikro2nd** [reply]
@quesada I'm curious: you've mentioned "backlinks" and "graph" a couple of times now as "things other note-taking apps add". Yet Zim has both (as plugins). Is there something deficient with the Zim implementations of those features? (Personally I have little use for the graph view (and it could be improved!) but I rely heavily on backlinks all the time.)

**Comment by vivasteve** [comment]
I see Zim as something between a note-taker and a wiki - gets harder to pigeon hole these apps - why should we even try ...

Zim is solid, reliable, does everything I need, an old friend

Have tried many of the newer electron-based apps - some of the features ... do I really need those?
In particular I do like Zettlr and Obsidian
Backlinks are cool - zim has backlinks, what's the problem
Graphs ... haven't personally found a need yet
Long live Zim
Jaap, if you do get tired of it all, we'll let you go with a warm feeling and much appreciation

**Reply by vivasteve** [reply]
PS - the Zim donation page is broken ...

**Comment by morenathan** [comment]
Do any of these alternatives just let me plop random files into my "notes" folder and put a link on my notes page and it's just there? To me that's the essence of future-proof. 

I have ages-old notes stored in Zim because I've been using it since Zim has been a thing. There was a time I toyed with replacing it with Leo editor but nah. Zim has been stable and uses an accessible text format that can directly access my filesystem without a bunch of garbage database files.

Show me any of the alternatives that don't think they need to parse, chop, funnel, filter, and mince my personal data through their system before saving it. Even if it's all processed and stored locally it still isn't accessible if you lose the GUI. At least with Zim, should it die some day, lowly old Bash could parse and use the files.

Pure markdown would be a nice alternative though :-) but I'm happy with what I have. I'm a simple boy afterall.

**Reply by vivasteve** [reply]
nicely said (:

**Reply by UnisTorvalds** [reply]
Here here!

**Comment by ZimTom** [comment]
Hello all,
I just spontaneously decided to sign up here after searching for an answer, thinking about posting an idea and during that ran into this thread and I thought it might be worthwhile to speak up as a very normal, ordinary, low end ZIM user.

Being an IT dinosaur of almost 62 and used to collect all kinds of personal information and related documents since the 1980s in anything from plain directory structures up to local website structures with handwritten static HTML, I recently fell in love with a personal Wiki created with ZIM.
I have to admit that I didn't run a proper market research when choosing ZIM, but still I tried quite a number of things, including mobile device apps etc.

My main arguments for ZIM are:

1. It is free for private use
2. It is platform independent and thus gives me flexibility in my upcoming platform decision. After my professional life has been more or less killed by the pandemic an my main system is not usable for Win11, I think of moving to Apple, because my private interests include photography and music. ZIM would allow me to take all my personal information across to a new platform very easily and with almost no risk of loosing anything.
3. It s working principle is simple and reliable
4. It takes minimal resources for operation as well as for storage, in fact so little that it still remains working while demanding applications like image processing eat up CPU and memory and most other things like office applications go down on their knees.
5. The functionality based on simple text files requires very little space in relation to the documents stored in my PIM structure, so that I can carry all my stuff with me requiring very limited hardware.
6. Because it is working based on text files and directory structures combined with a simple nomenclature it is particularly easy to get  existing information/document collections into my personal Wiki.

I know that this is by no means representative, but I have a strong preference for things following the KISS principle and thus providing a level of sustainability that is hard to find elsewhere. If I had used a personal Wiki like this much, much earlier I probably would not have lost so much information also during my professional career or at least would have saved a lot of work for making older information accessible again after the platform/application it was processed with was simply discontinued, became too expensive or was not accessible to me any more after changing job/employer.

Also in other areas I try to use simple, independent aplications and I hope ZIM will live forever - at least until I am no more able to use a computer, but from the picture I would assume that at least @jaap-karssenberg is young enough to survive me  :-)

Thanks a lot to him for the effort he puts in ZIM and the IMHO great app he keeps going with it.

**Comment by jaap-karssenberg** [comment]
Yes, I'm not close to 62 yet - but the downside is that my young kids keep
me away from working on zim in recent years - I still have the hope I have
much more time once I get to my 60s ;)

Also I'm using Zim daily in my professional life myself (engineering) so I
have a clear incentive to keep it up and running.

--- Jaap

**Reply by vivasteve** [reply]
Great to know about the incentive Jaap

**Reply by ZimTom** [reply]
I think having kids is probably the best reason in the world for not being able to get forward with things as fast you would like to :-)
At the same time I think it is an important source of motivation if you belong to the privileged category of people who are actually able to develop software tools themselves that they then can use for their job or even theier hobbies. So, I am not ashamed to admit being a bit selfish here and agree to what @vivasteve says, because the incentive makes it more likely that I as an ordinary user of ZIM can hope that it will last forever -well, at least for whatever "forever" means in my case :-)

Thanks a lot for doing this for you ... and for us !

**Reply by rzelnik** [reply]
This is really good news @jaap-karssenberg . Please consider accepting donations again. I am ready for supporting your work in small amounts, I just need you to allow me to do that. I would also recommend to consider [exit to community](https://www.youtube.com/watch?v=nRgZQHqb07A). Thank you for your work.

**Comment by jaap-karssenberg** [comment]
You're welcome!

Op zo 10 jul. 2022 11:25 schreef ZimTom ***@***.***>:

**Comment by mtill** [comment]
Hi!
can we (please) draft a new release from the current code base? the latest release was published quite some time ago and there are already some exciting new features implemented in our latest commits..
BR
mtill

**Reply by rzelnik** [reply]
Maybe it would be better to send this request into a separate thread. It has nothing to do with the topic of this discussion (The notetaker industry is moving really fast) and can be unnoticed.

**Comment by supertoothy** [comment]
I may have been using Zim for close to a decade. Still use it, everyday. Its always open on one monitor.

In no particular order, here are my opinionated reasons on why I will not switch away from zim to one of those new 'non-hierarchical associative' note taking apps anytime soon.

1. For longevity, it is not just enough to store in plain text. You also need to ensure that the structure of your files is usable at the filesystem level. Structure creates meaning, and Zim's design decision of creating a folder structure to match the notes structure ensures that even if Zim is gone, my structure still exists, with all my attachments in context to the notes. I don't just get all my notes in one mixed up folder.

2. Zim is so fast. I tried Obsidian, Logseq and they come nowhere near the speed at which Zim functions for note taking.

3. No block functions, thank heavens! I realised that the cost/benefits of having the granularity of block level referencing and embeds was actually making me do more pointless pruning and yak shaving, and in turn raising the complexity of the overall structure, turning the whole thing into a hyperlinked rabbithole. Zim on the other hand, lets me link to a page and to a Heading, if I need granular linking. That's generally enough for me.

4. I have 3 notebooks with sharing a total of about 15000 notes between them. Zim doesn't even stumble. I tried loading 700 txt files into Logseq - I could feel the lag. Maybe it's my old machine - but zim plays well with it. Great performance and wonderfully stable. I've lost notes in Onenote, Evernote, never in Zim.

I also use Zim as a file explorer, sort of. Because of how it creates nested directories, I store everything in it, documents, video files and working files from other software I use. Zim's Jump function lets me get to any file I want. When I need to backup all my work, I simply backup the entire Zim notebook and it brings everything with it.

To me Zim is feature complete and I cannot think of anything more I need from it. I'm saying this as someone who uses it everyday and all the time. 

The notes structures in my zim notebooks adapt to the twists and turns in my life smoothly. The hierarchical structure in the index acts as a mnemonic to the evolving structure of my work life and my interests. The hierarchy is not set in stone - I change it as my needs change.  

I cannot wrap my head around the whole roam type free-form interlinking without a centralized structure. It really doesn't work for me. 

I don't think I can move away from Zim. (I'm so paranoid that I may not have zim someday, that I regularly download the last stable version of Zim and store it in a HDD with a copy of the OS that worked well with it.)

**Reply by ZimTom** [reply]
Although I am far away from tis level of using ZIM, I can't do anything but perfectly agree.

Even today it is so nice to see positive examples for really resource efficiency like ZIM, keeping things simple without compromising functionality and safe-guard the accessability of the information by relying on base functionality like directoy structures combined with file/directory naming conventions. As I am still working on integrating information of an old document collection in mein ZIM Wiki I sometimes find it to be really helpful to be able to transfer whole branches of the old structure in my Wiki just by renaming and moving directories combined with quick and dirty creation of provisional TXT files, which can then be tidied up later. For tasks liekt this it would probably even be useful to have a UDL for ZIM in Notepad++.

And after I think of changing the platform sometime in the future - depending on the survival of my current PC - I am really that ZIM is available across platforms and uses the same base infrastructure, so that I would probably be able to move my entire Wiki from Windows to MacOS just by copying the whole tree from one machine to the other.

**Comment by menthalo37** [comment]
Just a few words : I'm using Zim for 7 years now (after having uses Rednotebook). I love text files and Zim loves them too. All of my colleagues are daily using One Note, which is not satisfying for me (I don't need synchro functionality , only stability and ability to quick search all of my mind corners somewhere on my hard disk...) I take notes about and during meeting, as well as I have task lists for main ways of my works (or small steps on projects).
I would be completly disappointed if Zim Notes would not work anymore. It's the first software I launch everyday, and all of my things are at the good place, waiting for me morning after morning.
Thanks for this tool.

**Comment by schiessle** [comment]
I wouldn't be that pessimistic. I tried many note taking tools over the last years. My main motivation was always to find something with a better mobile experience. But I always come back to Zim because at the end the features and the UX where more important than a mobile app. I still hope that at some point Zim will migrate to Markdown, which would solve the lack of mobile apps immediately. But otherwise I hope Zim exists as long as I need note taking apps, thanks a lot @jaap-karssenberg for the great work!

**Reply by hsandt** [reply]
I also mainly use Zim on PC for a faster overall flow. Most of the time it just does the job.

However, I also started using Joplin just to take notes on mobile (also because they were working on the new WYSIWYG editor that supports tables), and now I have my notes split into the two apps...

I can at least consult my Zim notes read-only from my Dropbox on mobile, and except for pictures and hyperlinks, the formatting is OK and readable. I suppose I *could* also edit notes by using Dropsync? I wish Dropbox would allow me to edit them live on the web, but recently the Dropbox Text Editor has become read-only and I'm not sure why.

I would I could switch completely to one app or the other, but the limitations mentioned in other posts or that I found out myself (Zim: Table is only supported as plugin and not great UX => easier to create pseudo-tables with tab, but alignment keeps breaking, various Jump to issues VS Joplin: encryption => harder to search through files as raw text, more reliance on Joplin app; media database => cannot simply open local folder containing pictures to resize them; Dropbox file change message spam while Joplin is open).

I just started Obsidian for local usage on PC but mobile looks good too, however I'd need to see how it can stand massive picture insertion (hopefully adding the files locally, very close to the md files, like Typora?) and backlinking.

Honestly I wouldn't mind that much if Zim was not adding new features but solidified what it already has: reliable tab indent would allow to simulate tables, stable Jump to would make navigation super fast, and then maybe if on my side I use Dropsync I'll even be able to edit Zim notes on mobile. Only issue remaining is inserting pictures, which I do a *lot* in Joplin, sometimes directly from my camera.

**Comment by rzelnik** [comment]
I just started to use [AnyType](https://anytype.io/) as my likely future replacement of Zim. I like its network way of linking documents together. I also like its database-like query capabilities for listing, sorting and filtering documents similar to Notion. Actually it is a local-first, cloud-independent Notion alternative. It also has a mobile app and supports synchronization through app developer's cloud or any sync tool of your choice. It works well with markdown (import-export, copy-paste, shortcuts). Unfortunately I don't know any clever way of data migration, so I will do it manually page by page.

**Reply by hsandt** [reply]
I heard about it when it was in closed alpha, but it's in open beta now! I'm having a look at it.

**Reply by rand00** [reply]
I just watched [a demo](https://youtu.be/3_7OcgHY49Y?t=601) - and I see why one could like this featurewise - but these electron apps always fail brutaly on their performance - which is so obvious here. The same was the case when I evaluated Logseq at some point. For me, zim hits the spot in a different way, and I have no need for switching.

Btw. if you are interested in this kind of tool, there is also [routine.co](https://routine.co/).

**Comment by fsteffenhagen** [comment]
I used Zim for a couple of years as a personal knowledgebase and for taking journal type notes for work. One of the main features I liked about Zim is the dead simple note hierarchy as files and directories which allows to easily manage attachments for notes and pages.
I'm currently using Obsidian for a couple of features that better suit my use cases (Markdown editing, mermaid diagrams) but I would consider moving back in favour of having a much simpler app.

**Comment by sojusnik** [comment]
In case somebody is interested to have a look at Obsidian, [here](https://github.com/obsidianmd/obsidian-importer/issues/39) are some tools listed for an automated migration.

**Reply by mikro2nd** [reply]
I've tried at least 4 times to like Obsidian, and kept going back to Zim. :)

**Comment by deminimis** [comment]
Zim can't really compete with more commercial programs like Obsidian in terms of features. But Obsidian uses electron, is bulky, it's slow. The two advantages to Obsidian are the huge list of plugins and it's more beautiful. 

I use Zim because it's faster. I don't need extra features. I just need front-end to manage directories full of text documents and photos and .pdf with a spell-check and a search function. Something I can completely firewall and store in a Veracrypt vault. 

In my opinion, Zim should not try to compete in features. It should trim as many features as possible and make them into optional plugins. Making the program as lightweight and fast as possible, without any glitches. And cross-platform if it wants to gain a lot of popularity.

**Reply by UnisTorvalds** [reply]
100% agree!!

**Comment by vivasteve** [comment]
@jonsnow231 
"And cross-platform
already cross-platform - versions available for linux, win, mac

**Reply by rzelnik** [reply]
Nowadays cross-platform means mobile platforms (Android, iOS) and desktop platforms (Win, Mac, Linux).

**Comment by vivasteve** [comment]
OK, point taken, not a mobile user here

**Comment by peter88213** [comment]
I have been using Zim for about ten years and am so satisfied with the currently installed version 0.69.1 that I still see no reason to update to the most recent version. At this point I would like to thank @jaap-karssenberg for this great piece of software!

In between I had a look at Obsidian and tried it out with my extensive standard notebook, for which I wrote a [conversion script](https://github.com/peter88213/zim2obsidian). I came to the conclusion that I like the concept of the Zim wiki pages better than Obsidian's organized file collection.

A huge advantage of Zim is that you can start it with command line parameters, which even allows you to switch pages from outside the process while it is running. This feature allows the connection to other software applications such as my own [novelibre](https://github.com/peter88213/novelibre) novel organizer, for which I am currently developing a [Zim connector plugin](https://github.com/peter88213/nv_zim).

**Reply by ZimTom** [reply]
Due to being locked out from my ZIM Wiki at the moment after having updated to MacOS 15 (currently 15.2) I got alarmed and also took a look at Obsidian. Yes, there are are some really nice features and it it is really cross-platform including mobile devices, but all in all I find it pretty overloaded, navigation based on an index as knowm from ZIM is really awkward, because of the different philosophy (a file being a file versus being a "node" or "object". The conversion from ZIM to markdown directly under ZIM worked fine, so that the content is kept, but to make the conversion result work compftable under Obsidian would require extensive manual rebuilding, so I am hesitant and hope that there will be an update coming out soon that supports MacOS 15 and that I am notrunning in the same problem next year again, when the Apple Update cycle takes its next lap.

**Reply by peter88213** [reply]
> but to make the conversion result work compftable under Obsidian would require extensive manual rebuilding,

That is the job of my conversion script mentioned above. The way I use Zim, the conversion was easy to do, but as the user feedback shows, there are cases where the exported markdown files do not reflect everything that can be displayed under Zim.

**Reply by ZimTom** [reply]
I can understand that the way ZIM is used has a a bi impact on how well a ZIM markdown export would be usable or at least easy to restructure. In my case I have a relatively granular structure following the philiospohy of a book index with lots of pages whose only function is to provide a bracket/grouping for some subordinate pages.

From what I saw, Obsidian as it is benefits if someone has been working with a relatively flat structure in ZIM.

An easy way to make migration to Obsidian basically a no-brainer would be to provide an extra view in the provided tree structure.
Today it shows basically an explorer / finder type view listing sub-folders within a folder in alphabetical order first, followed by the files in the folder. The only thing that would be needed to make converted ZIM Wikis easily usable in the way known  being used to olders in the tree view dissapear and you would have a list of files with indents according to th pragraph structure

**Reply by peter88213** [reply]
Zim's Markdown export has exactly the same folder structure as the wiki. I don't see any need to restructure the whole thing. The conversion to Obsidian by my script is essentially just a matter of aligning the file names with the page titles and adapting the advanced formatting, e.g. horizontal rulers, checkboxes, tags, and verbatim text. 

The overview pages, which serve as “nodes” in the wiki, are likewise available. If the links to the subpages have already been created in the wiki, they are also accessible in Obsidian. In Obsidian's file tree, they do not represent the nodes, but are on the same level as the subdirectories, which makes navigation somewhat different than with Zim. 
With the help of the tabs, however, you can also navigate from the homepage via the links and open a new tab if you want to go down a level. 

Then there are various plugins that [create navigable tables of contents](https://github.com/Robin-Haupt-1/Obsidian-Map-of-Content), or [support custom file ordering](https://github.com/lukasbach/obsidian-file-order), and and it says there is also a plugin that provides something like index pages for the directories. 

If none of this helps, it is not a big deal to quickly create a script to create link pages in the desired form. Or you can write your own plugin. Working with Obsidian can become a hobby. Nevertheless, I don't particularly like it and stick with Zim.

**Reply by ZimTom** [reply]
Thanks a lot for your input. Unfortunately I belong to the kind of people that normally have no place on a platform like Github, because I am much further away from programming than probably the vast majority here. Thus, for people liek me it ACTUALL WOULD BE a big deal to write things like scripts or plugins myself :-)

As fas as the markdown exprt of ZIM is concerned, I was aware about the maintaining of the original structure because I tested it.
But I will take a look at the available plugins for Obsidian. Especially the custom file ordering thíng just for interest.

In the meantime @dehesselle was so kind to provide a workaround for the problem with MacOS 15 and my ZIM world is all fine again.

**Reply by peter88213** [reply]
> and my ZIM world is all fine again.

Then all's well. I didn't want to sound pretentious about scripting, but the advantage of text files (both Zim and Obsidian) is that you can process them even at beginner level if you learn one of the common scripting languages. 

With the plugins it's a different story, of course.

**Reply by ZimTom** [reply]
> I didn't want to sound pretentious about scripting

No problem, to me it didn't sound pretentious at all. It is normal that you are looking at a topic in a different way if you are standing right in it knee-deep compared to somebody floating on the surface :-)
The same happens to me regularly, too, when talking with others about my work or hobbies. Enthusiasm is a good trigger for acceleration in communication and sometimes you reach the speed limit of others without even realising it :-D

Ages ago I did quite a bit of programming (PLC program design and coding for the job, MIDI-related stuff for my hobby), but if you have to learn new things you also have to de-learm others to manage the memory, so these things got in off ... but it is always interesting to get in contact with it again these days - and leave the complex part to the pros.

**Reply by peter88213** [reply]
> if you have to learn new things you also have to de-learm others to manage the memory,

That sounds like a good idea. Run the garbage collector over it, then the defragmenter, and then get _Head First Python_ for the perfect start into the next year. 
;-)

**Comment by morenathan** [comment]
I am personally finding myself spending more time with my Galaxy Tab. It's not a notebook for sure, so had to find a solution that satisfies the similarities of Zim. For some time now I've been using Markor, which syncs between my phone and tablet fine using a folder structure and markdown.



It has some support for Zim/Text pages but the linking is always wonky for me. So I've somewhat on my longtime running Zim usage and am slowly translating everything to markdown.



Markdown has a great set of tools and is extensible pretty easily. I haven't checked on that conversion in Zim yet for awhile.






---- On Sat, 07 Dec 2024 06:56:21 -0600 peter88213 ***@***.***> wrote ---





I have been using Zim for about ten years and am so satisfied with the currently installed version 0.69.1 that I still see no reason to update to the current version. At this point I would like to thank https://github.com/jaap-karssenberg for this great piece of software!

In between I had a look at Obsidian and tried it out with my extensive standard notebook, for which I wrote a https://github.com/peter88213/zim2obsidian. I came to the conclusion that I like the concept of the Zim wiki pages better than Obsidian's organized file collection.

A huge advantage of Zim is that you can start it with command line parameters, which even allows you to switch pages from outside the process while it is running. This feature allows the connection to other software applications such as my own https://github.com/peter88213/novelibre novel organizer, for which I am currently developing a https://github.com/peter88213/nv_zim.

**Reply by ZimTom** [reply]
Well, I am far from being an expert, but the reasons menttioned in the reply to @peter88213 (see last post above) I just tried the markdown export from ZIM and it seems to work really fine. The only thing I spotted so far is that for one headline format (I think it was H2) the conversion adds the markdown format tags but doesn't remove the ZIM format tags, so that you might have to delete some "-" here and there.

**Comment by bbekir** [comment]
I have tried almost all major notetaking apps but  turned back  every time.
Markdown-based applications often add additional load. The functions they all add make it impossible to move between them.

In other applications which do not organize data in the file system as like Zim and use their own file format or database, in addition to the risk of losing notes in a possible failure, it is not possible to search and work with files at the file system level . 

So what's the problem?
Attachments; GREAT :)))
Syntax highlighter; could be better
Table editor; ..eerm ok
Tags; ok
Editor capabilities; sufficient
Search capabilities; should have a toolbar
Mobile support: i dont care
Sync: i dont care
Modern gui: i dont care
Adding multiple template : missing !
Adding property to notes (category etc) : missing !!!

I think category is must and can be implemented without manipulating phylosophy

At past i have tried to add features like categories to notes. My plan was as follows;
Splitting tags into two with special characters; something like @category|personal
Creating multiple templates according to the default properties and adding backlincks at the bottom of the pages : technical notes  etc
Creating a master template for categories and displaying backlinks on the page

Unfortunately, special characters like  | & : are not allowed in tags. Also if it is possible to create an extra template, I can't do it.

I wish I were a software developer :((

**Reply by Nick222** [reply]
> Adding property to notes (category etc) : missing !!!

But why user cannot use tags for this?

Or make file descript.ion for note - and write in it all comments etc.

**Reply by peter88213** [reply]
> Splitting tags into two with special characters; something like `@category|personal`

Do you actually need a "category" prefix? For me, the tag _is_ the category.
If necessary, you could use the underscore as a separator.

**Reply by ZimTom** [reply]
It is good to hear - sorry, read - that someone has tried out so many notetakers and is still convinced of staying with ZIM. From the feeling I think the same, but of course beliving / feeling is not the same.

When starting to get into Obsidian, it became obvious that there are some really nice feature and it looks nice, but compared with ZIM it really looks overloaded and ZIM certainly is much more lightweight and down to earth, concentrating on the crucial things needed.

Cross-platform is also for me not a great concern, at least if it comes to mobile devices. That said, cross-platform working at least cross computer OS boundaries (Windows, MacOS, ...) has turned out to be a life saver for me, because I was locked out from my Wiki due to an update of MacOS to version 15.x and now I can - hopefully provisionally and short term - use my Windows dinosaur to use my Wiki and keep it alive.
